sklearn/sklearn-template/template/trainer/task.py

```diff
@@ -15,28 +15,85 @@
 
 """Executes model training and evaluation."""
 
-
+import os
+import time
 import argparse
 import logging
 import sys
+from urllib.parse import urlparse
 
 from sklearn import model_selection
-from tensorflow import gfile
+from sklearn.externals import joblib
+
+import numpy as np
+
+import hypertune
+
+from google.cloud import storage
+
+from trainer.input_util import read_from_bigquery
+from trainer.model import get_estimator
+from trainer.metadata import MODEL_FILE_NAME_PREFIX
+from trainer.metadata import METRIC_FILE_NAME_PREFIX
+from trainer.metadata import DUMP_FILE_NAME_SUFFIX
+
+
+def _upload_to_gcs(local_path, gcs_path):
+  """
+  Upload local file to Google Cloud Storage
+
+  Args:
+    local_path: (string) Local file
+    gcs_path: (string) Google Cloud Storage destination
+
+  Returns:
+    None
+  """
+  storage_client = storage.Client()
+  parse_result = urlparse(gcs_path)
 
-from trainer import input_util
-from trainer import model
+  # Parse bucket name
+  gcs_path = parse_result.path
+  bucket_name = parse_result.hostname
+  bucket = storage_client.get_bucket(bucket_name)
```

Comment:
I'd recommend the `tensorflow.gfile` approach as it's simpler (see my comments in PR #4 ), unless there is an advantage you see in using `storage`.

Reply:
There are two reasons that I think we shouldn't use tf.gfile
1. All the other examples in sklearn/ folder don't use tf.gfile, I think we should align to them
2. If people choose to use sklearn, in a lot of cases I would assume that they don't have tensorflow experience or knowledge. Even just making use of tf.gfile may still confuse people.

Therefore, I remove the tensorflow dependency completely. What do you think?

Reply:
Thanks for the feedback. Here are my thoughts regarding your points:
1. I understand. We would be changing the convention then. We could let the repo owners decide if they would want to go with this approach or keep the existing one.
2. Right, actually `gfile` is more closely related to the Python `file` module than anything in `tensorflow`. So they wouldn't really need to know anything about TensorFlow at all, other than the fact that `gfile` exists and they could read more about it in: https://www.tensorflow.org/api_docs/python/tf/gfile

While it's a bit unfortunate that they placed a great file handling module `gfile` into `tensorflow` specifically, I think we should still use it because it reduces the complexity of reading/writing files from/to GCS. It takes about a couple of lines of code to do GCS file I/O with `gfile`.

Practically speaking, if you were a user (say a data scientist), and you learned about `gfile`, would you still want to use `storage` for simple GCS file I/O?

Also, I think it would be nice to get users thinking about `tensorflow` anyway, even though the main framework they're using is `sklearn`. :)

And we could use `from tensorflow import gfile` import convention so we don't need to propagate `tf` throughout the rest of the code.